### PR TITLE
Clean-up and document goto_trace.h/cpp [DOC-136]

### PR DESCRIPTION
**Pull request by romainbrenguier**
No functional changes: only removing unused code and documenting.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [na] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]

File: src/goto-programs/goto_trace.h [hunk 1]
```diff
@@ -24,12 +24,32 @@ Date: July 2005
 
 #include <goto-programs/goto_program.h>
 
-/*! \brief TO_BE_DOCUMENTED
- * \ingroup gr_goto_symex
-*/
+/// Step of the trace of a GOTO program
+///
+/// A step is either:
+///   - an assignement
```
Comment (peterschrammel) [inline src/goto-programs/goto_trace.h]: assignment

File: src/goto-programs/goto_trace.h [hunk 2]
```diff
@@ -186,14 +207,23 @@ class goto_tracet
...
+  /// Add rawLhs property to trace
   bool json_full_lhs;
+  /// Represent plain trace values in hex
   bool hex_representation;
+  /// Use prefix (`0b` or `0x`) for distinguishing the base of the
+  /// representation.
   bool base_prefix;
+  /// Show function calls in plain trace
   bool show_function_calls;
+  /// how original code in plain trace
```
Comment (peterschrammel) [inline src/goto-programs/goto_trace.h]: Show
Maybe: `in plain text trace`

**Review by peterschrammel — APPROVED** [review]

File: src/goto-programs/goto_trace.h [hunk 3]
```diff
@@ -24,12 +24,32 @@ Date: July 2005
...
+///   - a function call
+///   - a function return
+///   - a location (unused)
+///   - an output
+///   - an input
+///   - a declaration
+///   - a dead statement
+///   - a shared read (unused)
+///   - a shared write (unused)
+///   - a spawn satemement
```
Comment (edstenson) [inline src/goto-programs/goto_trace.h]:
```suggestion
///   - a spawn statement
```

**Review by edstenson — CHANGES_REQUESTED** [review]
Typo

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 20638d7cf2ec1abc01a68df32f6ca45c35ce3c81).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99755367
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 0c7e8d879b43338e52d4541caf05202fb1d0bd57).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99756516
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b88944db2873350d1c5c3eab22cb9fc901dc209c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99764171

File: src/goto-programs/goto_trace.h (hunk 1 above)
Comment (edstenson) [reply src/goto-programs/goto_trace.h]:
```suggestion
///   - an assignment
```

**Pull request merged**